src/libraries/System.Runtime/tests/System.Runtime.Tests/System/String.SplitTests.cs

```diff
@@ -634,6 +645,13 @@ public static void SplitCharArraySeparator(string value, char[] separators, int
             Range[] ranges = new Range[count == int.MaxValue ? value.Length + 1 : count];
             Assert.Equal(expected.Length, value.AsSpan().SplitAny(ranges, separators, options));
             Assert.Equal(expected, ranges.Take(expected.Length).Select(r => value[r]).ToArray());
+
+            // The SpanSplitEnumerator does not replicate the behaviour of splitting on a whitespace char when input char[] is null or empty.
+            if (count == int.MaxValue && options is StringSplitOptions.None && separators is { Length: > 0 })
```

Comment:
This check is _slightly_ suboptimal as it inadvertently also skips [these](https://github.com/dotnet/runtime/pull/104534/files#diff-2fc66ca55c078e629cf35edb1bd4e768e928f5f94b8b1c7b6b9d81422a91c78fL598) two test. I wasn't entirely sure how to proceed, I could write the condition to specifically include those two tests based on the exact inputs, but that seems wrong. The alternative would be splitting those two tests into a separate method. I am however specifically testing this scenario in the generic tests above.